x-pack/plugins/stack_connectors/public/connector_types/pagerduty/links_list.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  ActionParamsProps,
+  TextFieldWithMessageVariables,
+} from '@kbn/triggers-actions-ui-plugin/public';
+import { PagerDutyActionParams } from '../types';
+
+type LinksListProps = Pick<
+  ActionParamsProps<PagerDutyActionParams>,
+  'index' | 'editAction' | 'errors' | 'messageVariables'
+> &
+  Pick<PagerDutyActionParams, 'links'>;
+
+export const LinksList: React.FC<LinksListProps> = ({
+  editAction,
+  errors,
+  index,
+  links,
+  messageVariables,
+}) => {
+  const areLinksInvalid = Array.isArray(errors.links) && errors.links.length > 0;
+
+  return (
+    <EuiFormRow
+      id="pagerDutyLinks"
+      label={i18n.translate('xpack.stackConnectors.components.pagerDuty.linksFieldLabel', {
+        defaultMessage: 'Links (optional)',
+      })}
+      isInvalid={areLinksInvalid}
+      error={errors.links}
+      fullWidth
+    >
+      <EuiFlexGroup direction="column" data-test-subj="linksList" gutterSize="s">
+        {links &&
+          links.map((link, currentLinkIndex) => (
+            <EuiFlexItem data-test-subj="linksListItemRow">
+              <EuiSpacer size="s" />
+              <EuiFlexGroup key={`my-key-${currentLinkIndex}`}>
```

Comment:
I found the `my-key` prefix too generic. What about something more specific to the functionality?